src/routes/deleteProduct.routes.js

```diff
@@ -0,0 +1,71 @@
+import express from 'express';
+import deleteProduct from '../controllers/deleteProduct.controller';
+import { authorizeUser } from '../middlewares/userRoles.middleware';
+import RoleCheck from '../middlewares/user.Checkrole';
+
+const router = express.Router();
+
+router.delete('/deleteProduct/:id', authorizeUser, RoleCheck(['merchant', 'admin']), deleteProduct);
```

Comment:
add a middleware validator for the product ID param

Reply:
authorizeUser and RoleCheck are middleware